File: compare_image.cpp
#include <opencv2/opencv.hpp>

int main() {
    // Load images
    cv::Mat referenceImage = cv::imread("android.png");
    cv::Mat testImage = cv::imread("android1.png");

    // Check if the images are loaded successfully
    if (referenceImage.empty() || testImage.empty()) {
        std::cerr << "Error loading images." << std::endl;
        return -1;
    }

    // Resize images to a common size
    cv::resize(referenceImage, referenceImage, testImage.size());

    // Convert images to grayscale
    cv::Mat referenceGray, testGray;
    cv::cvtColor(referenceImage, referenceGray, cv::COLOR_BGR2GRAY);
    cv::cvtColor(testImage, testGray, cv::COLOR_BGR2GRAY);

    // Calculate Structural Similarity Index
    cv::Scalar mssim = cv::quality::QualityBRISQUE::compute(referenceGray, testGray);

    // Display result
    double ssimIndex = (mssim[0] + mssim[1] + mssim[2]) / 3.0;

    if (ssimIndex >= 0.95) {
        std::cout << "Images are similar. SSIM Index: " << ssimIndex << std::endl;
        // Optionally, you can display the images here.
    } else {
        std::cout << "Images are not similar. SSIM Index: " << ssimIndex << std::endl;
    }

    return 0;
}

File: compare_image.py
import imageio.v2 as imageio
import dssim

def image_similarity(image_path1, image_path2, threshold=0.95):
    # Load images
    image1 = imageio.imread(image_path1)
    image2 = imageio.imread(image_path2)

    # Calculate Structural Similarity Index
    ssim_index = dssim.calculate_ssim(image1, image2)

    # Display result
    if ssim_index >= threshold:
        print("Images are similar. SSIM Index:", ssim_index)
        # Optionally, you can display the images here.
    else:
        print("Images are not similar. SSIM Index:", ssim_index)

# Example usage
reference_image_path = "android.png"
test_image_path = "android1.png"

image_similarity(reference_image_path, test_image_path)

